tests/test_utils.py

```diff
@@ -0,0 +1,151 @@
+import lsst.ip.isr as ipIsr
+import lsst.afw.image as afwImage
+from lsst.afw.geom import Box2I, Point2I, Extent2I
+import lsst.log as lsstLog
+
+import lsst.cp.pipe.utils as cpUtils
+
+
+class UtilsTestCase(lsst.utils.tests.TestCase):
+    """A test case for the utility functions for cp_pipe."""
+
+    def setUp(self):
+
+        mockImageConfig = isrMock.IsrMock.ConfigClass()
+
+        # flatDrop is not really relevant as we replace the data
+        # but good to note it in case we change how this image is made
+        mockImageConfig.flatDrop = 0.99999
+        mockImageConfig.isTrimmed = True
+
+        self.flatExp = isrMock.FlatMock(config=mockImageConfig).run()
+        (shapeY, shapeX) = self.flatExp.getDimensions()
+
+        self.rng = np.random.RandomState(0)
+        self.flatMean = 1000
+        self.flatWidth = np.sqrt(self.flatMean)
+        flatData = self.rng.normal(self.flatMean, self.flatWidth, (shapeX, shapeY))
+        self.flatExp.image.array[:] = flatData
+
+    def test_countMaskedPixels(self):
+        exp = self.flatExp.clone()
+        mi = exp.maskedImage
+        self.assertEqual(cpUtils.countMaskedPixels(mi, 'NO_DATA'), 0)
+        self.assertEqual(cpUtils.countMaskedPixels(mi, 'BAD'), 0)
+
+        NODATABIT = mi.mask.getPlaneBitMask("NO_DATA")
+        noDataBox = Box2I(Point2I(31, 49), Extent2I(3, 6))
+        mi.mask[noDataBox] |= NODATABIT
+
+        self.assertEqual(cpUtils.countMaskedPixels(mi, 'NO_DATA'), noDataBox.getArea())
+        self.assertEqual(cpUtils.countMaskedPixels(mi, 'BAD'), 0)
+
+        mi.mask[noDataBox] ^= NODATABIT  # XOR to reset what we did
+        self.assertEqual(cpUtils.countMaskedPixels(mi, 'NO_DATA'), 0)
+
+    def test_parseCmdlineNumberString(self):
+        parsed = cpUtils.parseCmdlineNumberString('1..5')
+        self.assertEqual(parsed, [1, 2, 3, 4, 5])
+
+        parsed = cpUtils.parseCmdlineNumberString('1..5:2^123..126')
+        self.assertEqual(parsed, [1, 3, 5, 123, 124, 125, 126])
+
+        parsed = cpUtils.parseCmdlineNumberString('12^23^34^43^987')
+        self.assertEqual(parsed, [12, 23, 34, 43, 987])
+
+        parsed = cpUtils.parseCmdlineNumberString('12^23^34^43^987..990')
+        self.assertEqual(parsed, [12, 23, 34, 43, 987, 988, 989, 990])
+
+    def test_checkExpLengthEqual(self):
+        exp1 = self.flatExp.clone()
+        exp2 = self.flatExp.clone()
+
+        self.assertTrue(cpUtils.checkExpLengthEqual(exp1, exp2))
+
+        visitInfo = afwImage.VisitInfo(exposureTime=-1, darkTime=1)
+        exp2.getInfo().setVisitInfo(visitInfo)
+        self.assertFalse(cpUtils.checkExpLengthEqual(exp1, exp2))
+
+        with self.assertRaises(RuntimeError):
+            cpUtils.checkExpLengthEqual(exp1, exp2, raiseWithMessage=True)
+
+    def test_validateIsrConfig(self):
+
+        mand = ['doAssembleCcd']
+        forb = ['doFlat', 'doFringe']
+        des = ['doBias', 'doDark']
+        undes = ['doLinearize', 'doBrighterFatter']
```

Comment:
`mand forb des`. The new boardgame from Fantasy Flight!

Please give them non-abbreviated names.

Reply:
This was necessary for the line lengths. I originally had it like that, but it was _such_ a mess. I don't think this is worth it. I've added comments to the lines instantiating these to make it clear what they are though, I think that should suffice.

Reply:
"This was necessary for the line lengths." That's not good. Looking at the lines below, changing e.g. `cpUtils.validateIsrConfig(task, mand + ['test'], forb, des, undes)` to:

```
cpUtils.validateIsrConfig(task,
                          mandatory + ['test'],
                          forbidden,
                          desirable,
                          undesirable)
```

seems like a win on several fronts (full words, breaking up long parameter list).

Reply:
Whatever floats your goat. 🐐